File: backend/src/main/java/com/ecobook/dto/UsuarioDTO.java
package com.ecobook.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.Set;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UsuarioDTO {
    private String id;
    private String email;
    private String nome;
    private String whatsapp;
    private String cidade;
    private String bairro;
    private String instituicao;
    private Boolean perfilCompleto;
    private Boolean consentimentoIa;
    private String role;
    private Set<String> necessidadesAcademicas;
    private LocalDateTime criadoEm;
    private LocalDateTime atualizadoEm;
}

File: EcoBookAiAndroid/app/src/main/kotlin/com/ecobook/dto/UsuarioDTO.kt
package com.ecobook.dto

data class UsuarioDTO(
    val id: String,
    val email: String,
    val nome: String,
    val whatsapp: String? = null,
    val cpf: String? = null,
    val cidade: String? = null,
    val bairro: String? = null,
    val instituicao: String? = null,
    val fotoPerfilUrl: String? = null,
    val perfilCompleto: Boolean,
    val consentimentoIa: Boolean = false,
    val necessidadesAcademicas: Set<String> = emptySet(),
    val role: String = "USER"
)
